### PR TITLE
Add a test for turning normal enum into extensible

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: test/TestProjects/TypeSchemaMapping/Customized/NonExtensibleEnumTurnedExtensible.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+using Azure.Core;
+
+namespace NamespaceForEnums
+{
+    [CodeGenModel("NonExtensibleEnum2")]
+    internal partial struct NonExtensibleEnumTurnedExtensible
+    {
+    }
+}
```
Comment (ShivangiReja) [inline test/TestProjects/TypeSchemaMapping/Customized/NonExtensibleEnumTurnedExtensible.cs:12]: The intention of adding above is to convert normal enum into extensible enum? If yes, then I don't think you needed the above in order to generate extensible enum.  I saw that before and I believe below spec will generate extensible enum even if you don't mention the above. 

```
"NonExtensibleEnum": {
      "description": "More Letters",
      "type": "string",
      "enum": [
        "A",
        "B",
        "C"
      ]
    }
```

Comment (ShivangiReja) [reply test/TestProjects/TypeSchemaMapping/Customized/NonExtensibleEnumTurnedExtensible.cs:12]: Please let me know if I'm missing anything here.

Comment (pakrym) [reply test/TestProjects/TypeSchemaMapping/Customized/NonExtensibleEnumTurnedExtensible.cs:12]: Yes, this test is for ability for developers to switch between generated enum kinds without changing swaggers. It's similar to how we can rename properties, change propetry types, etc.

Comment (ShivangiReja) [reply test/TestProjects/TypeSchemaMapping/Customized/NonExtensibleEnumTurnedExtensible.cs:12]: Yeah, but what I was trying to say that `"modelAsString": false` is missing in the spec, if we don't mention `"modelAsString"` in the spec, it by default generates extensible enum. So there is no point of below:
```
[CodeGenModel("NonExtensibleEnum2")]
    internal partial struct NonExtensibleEnumTurnedExtensible
    {
    }
```

Comment (pakrym) [reply test/TestProjects/TypeSchemaMapping/Customized/NonExtensibleEnumTurnedExtensible.cs:12]: Yep, you are right. Fixing.

**Pull request merged**